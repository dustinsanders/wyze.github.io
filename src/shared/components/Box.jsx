// @flow

import type { Style, VNode } from '../../types'
import { css, media, select } from 'glamor'
import { flexGrow, level } from '../styles'
import { large } from '../styles/breakpoints'
import Title from './Title'
import makeRGBA from '../helpers/makeRGBA'
import preact from 'preact'

type Props = {
  children?: VNode,
  title?: string,
  wrap?: boolean,
  childStyle?: Style,
}

const shadow: string = makeRGBA(0.1)
const box: Style = css(
  {
    backgroundColor: '#fff',
    borderRadius: 5,
    boxShadow: `0 2px 3px ${shadow}, 0 0 0 1px ${shadow}`,
    display: 'block',
    padding: '1em',
  },
  select('& + &', {
    marginTop: '2em',
  }),
  media(large, select('& + &', {
    marginTop: '4em',
  })),
)
const flexWrap: Style = css({ flexWrap: 'wrap' })

const Box = ({ children, title, wrap, childStyle }: Props): VNode => (
  <div {...box}>
    {title ? <Title title={title} /> : <span />}
    <div {...css(level, flexGrow, wrap && flexWrap, childStyle)}>
      {children}
    </div>
  </div>
)

Box.defaultProps = {
  childStyle: {},
  children: null,
  title: null,
  wrap: false,
}

export default Box
