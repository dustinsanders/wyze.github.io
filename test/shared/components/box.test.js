// @flow

import Box from '../../../src/shared/components/Box'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const childStyle = {
  color: 'rebeccapurple',
}

const children = (
  // eslint-disable-next-line react/jsx-filename-extension
  <div>I am a child.</div>
)

const title = 'I am a title.'

describe('<Box />', () => {
  it('renders', snapshot.bind(null, Box))
  it('renders children', snapshot.bind(null, Box, { children }))
  it('renders title', snapshot.bind(null, Box, { title }))
  it('renders flex wrap', snapshot.bind(null, Box, { wrap: true }))
  it('renders passed styles', snapshot.bind(null, Box, { childStyle }))
})
