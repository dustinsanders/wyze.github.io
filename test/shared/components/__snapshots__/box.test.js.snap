// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders 1`] = `
"<div data-css-17z6r2c>
  <span></span>
  <div data-css-1w4mng5></div>
</div>"
`;

exports[`<Box /> renders children 1`] = `
"<div data-css-17z6r2c>
  <span></span>
  <div data-css-1w4mng5>
    <div>I am a child.</div>
  </div>
</div>"
`;

exports[`<Box /> renders flex wrap 1`] = `
"<div data-css-17z6r2c>
  <span></span>
  <div data-css-1xbb4u></div>
</div>"
`;

exports[`<Box /> renders passed styles 1`] = `
"<div data-css-17z6r2c>
  <span></span>
  <div data-css-1gmhrpi></div>
</div>"
`;

exports[`<Box /> renders title 1`] = `
"<div data-css-17z6r2c>
  <Title title=\\"I am a title.\\"></Title>
  <div data-css-1w4mng5></div>
</div>"
`;
